### DISCUSSION TITLE
Flame and Box2D and functional

### DISCUSSION THREAD

**Discussion by pedia** [flame-engine/flame#694] [General]
Usually, when building games, you will encounter movement, rotation, and collision. These functions in the flame are not so powerful now, but these functions have been tested in Box2D for a long time and have excellent efficiency. 
So I was thinking:
- Does the focus of Flame is a bit duplicated with Box2D?
- Should Flame focus on Image, Sprite, AnimationSprite, Map, Level Data, and assets of bunding?

Or there how many games these do not need moving, rotation, or collide?

Some thoughts and questions from newcomers of game dev.

Thanks.

**Comment by erickzanardo** [comment]
@pedia Flame provides box2d support already

https://github.com/flame-engine/forge2d
https://github.com/flame-engine/flame_forge2d

**Reply by pedia** [reply]
Yes, I have read a little of Forge2d.

To be more precise, 
When dealing with the rotation, movement, and collision of `SpriteComponent`, Is it more appropriate to directly use the `Body` in `Forge2D`?

**Reply by erickzanardo** [reply]
It really depends on what you need, our components system offer some basics features that will be good enough for some games, but for other games that needs more advanced features, then you go with our box2d implementation.

**Reply by spydon** [reply]
You can use a `SpriteComponent` on top of a Forge2D body with the `PositionBodyComponent` in `flame_forge2d` if that is what you are looking for.

**Comment by pedia** [comment]
The success of Unity is not accidental, Texture2D, AudioClip, and Assets Packaging are more suitable for developers.
